seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/utils/CatalogUtils.java

```diff
@@ -201,7 +205,7 @@ public static CatalogTable getCatalogTable(Connection connection, TablePath tabl
                 catalogName);
     }
 
-    public static CatalogTable getCatalogTable(ResultSetMetaData resultSetMetaData)
+    public static CatalogTable getCatalogTable(ResultSetMetaData resultSetMetaData, String sqlQuery)
```

Comment:
Incoming parameters should be used for logical processing

```java
public static CatalogTable [your call method](ResultSetMetaData resultSetMetaData, String sqlQuery) {
    try {
        return getCatalogTable(resultSetMetaData);
    } catch (Exception e) {
        throw CommonError.getCatalogTableWithUnsupportedType(..., sqlQuery);
    }
}

public static CatalogTable getCatalogTable(ResultSetMetaData resultSetMetaData) {
    ...
}
```

Reply:
I think error handling is also a very important part of the logic. If we don't do this, we need to perform try catch everywhere where this method is called, and then require the caller to return appropriate error information, which is very unreasonable.

Reply:
👌